core/src/main/java/com/valtech/aem/saas/core/http/response/TypeaheadDataExtractionStrategy.java

```diff
@@ -0,0 +1,60 @@
+package com.valtech.aem.saas.core.http.response;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.MapUtils;
+
+/**
+ * A strategy for extracting typeahead options.
+ */
+@RequiredArgsConstructor
+public final class TypeaheadDataExtractionStrategy implements SearchResponseDataExtractionStrategy<List<String>> {
+
+  private static final String TYPEAHEAD_SOLR_INDEX_FIELD_NAME_PREFIX = "autocomplete_text_";
+
+  private final String language;
+
+  @Override
+  public String propertyName() {
+    return FacetCounts.PN_FACET_COUNTS;
+  }
+
+  @Override
+  public Optional<List<String>> getData(JsonElement response) {
+    return Optional.ofNullable(response)
+        .filter(JsonElement::isJsonObject)
+        .map(JsonElement::getAsJsonObject)
+        .map(r -> r.getAsJsonObject(propertyName()))
+        .map(jsonObject -> new Gson().fromJson(jsonObject, FacetCounts.class))
+        .map(this::getTypeaheadResults);
+  }
+
+  private List<String> getTypeaheadResults(@NonNull FacetCounts facetCounts) {
+    return Optional.ofNullable(facetCounts.getFacetFields())
+        .filter(MapUtils::isNotEmpty)
+        .map(map -> map.get(getTypeaheadSolrIndexFieldName()))
+        .map(this::getTypeaheadItems)
+        .orElse(Collections.emptyList());
+  }
+
+  private List<String> getTypeaheadItems(@NonNull List<Object> facetFieldsList) {
+    List<String> items = new ArrayList<>();
+    for (int i = 0; i < facetFieldsList.size(); i++) {
```

Comment:
this looks weired. can you provide an example? I don't get why odd elements behave differently

Reply:
check file -> core/src/test/resources/__files/search/typeahead/success.json for example

Reply:
thank you! weired solr response :) 
can you extract the if condition into a method `isTypeaheadTextOption(facetFieldsList.get(i))` so it get's more readable?